adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -65,61 +73,167 @@ private GenericContainer<?> createAndStartCollector(String configFilePath, Strin
 
-        validateLogs(logStreamName , "/logs/testingJSON.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/testingJSON.log");
+        validateLogs(logStreamName , logFilePaths, true);
         collector.stop();
     }
 
-    void validateLogs(String testLogStreamName, String logFilePath) throws Exception {
-        var file = new File(logFilePath);
+    @Test
+    void testCollectorRestartStorageExtension() throws Exception {
+        String logStreamName = "storageExtension-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-storageExtension.yaml", logStreamName);
+        File tempFile = new File(logDirectory.toString(), "storageExtension.log");
+        Thread.sleep(5000);
+
+        FileWriter fileWriter = new FileWriter(tempFile);
+        try {
+            fileWriter.write("First Message, collector is running" + "\n");
+            fileWriter.write("Second Message, collector is running" + "\n");
+            fileWriter.write("Third Message,  collector is running" + "\n");
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        fileWriter.flush();
+
+        List<String> logFilePaths = new ArrayList<>();
+        String expectedLogPath = logDirectory.toString();
+        logFilePaths.add(expectedLogPath + "/storageExtension.log");
+
+        validateLogs(logStreamName , logFilePaths, false);
+
+        collector.stop();
+
+        Thread.sleep(5000);
+
+        // write to the file when collector is stopped
+        try {
+            fileWriter.write("First Message after collector is stopped" + "\n");
+            fileWriter.write("Second Message after the collector is stopped" + "\n");
+            fileWriter.write("Third Message after the collector is stopped" + "\n");
+            fileWriter.flush();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        fileWriter.close();
+       //restarting the collector
+        collector.start();
+
+        validateLogs(logStreamName , logFilePaths, false);
+        collector.stop();
+    }
+
+    @Test
+    void testFileRotation() throws Exception {
+        String logStreamName = "fileRotation-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-fileRotation.yaml", logStreamName);
+
+        Thread.sleep(5000);
+
+        // Create and write data to File A
+        File tempFile = new File(logDirectory.toString(), "testlogA.log");
+        try (FileWriter fileWriter = new FileWriter(tempFile)) {
+            fileWriter.write("Message in File A" + "\n");
+            fileWriter.flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        Thread.sleep(5000);
```

Comment:
you van validate the log entries so far here so that you don't need to add a sleep. that will block in a safe way before you move with the rest of the tests.

Reply:
I agree that validation can be done, but the thread here is making sure to give enough time for the file operations to be done. So even if we do the validation without the thread, since the log file is not generated it would lead FileNotFoundException.

Reply:
Is writing to the file a blocking operation? Why do you need to wait? What file operations are you referring to?

Reply:
what operations needs to be done?

Reply:
File operations i meant here are to create and write data to File A etc. it takes considerable time for creating and writing to file.
So without the thread.sleep the collector is not able to parse log line, since it seems the file testlogA.log didn't have the line written to it yet.

```
        java.lang.AssertionError: 
        Expecting actual:
          ["Message in renamed file - line 1",
            "Message in renamed file - line 2",
            "Message in renamed file - line 3"]
        to contain exactly in any order:
          ["Message in File A",
            "Message in renamed file - line 1",
            "Message in renamed file - line 2",
            "Message in renamed file - line 3"]
        but could not find the following elements:
          ["Message in File A"]
```

Reply:
that is why I suggested that you add a `validateLogs` instead of a `Thread.sleep` to validate only "Message in File A". It will block until the lines are sent to cwlogs. This will guarantee that it is safe to continue with the tests.

Reply:
Committed the changes.